coding-standards-and-naming.rst

```diff
@@ -238,7 +238,8 @@ library. The application of common sense is recommended.
    (``int8_t``, ``int16_t``, ``int32_t``, ``int64_t`` and their unsigned
    counterparts) are supposed to be available.
 
-#. Functions that are given pointers to resources MUST NOT free them.
+#. Functions that are given pointers to resources SHOULD NOT free them, unless
+   this is documented.
```

Comment:
This really depends on the function and logic. I think this whole think should be removed as well as it's context dependent.